modules/mauth-authenticator-akka-http/src/test/scala/com/mdsol/mauth/akka/http/MAuthDirectivesSpec.scala

```diff
@@ -139,7 +178,205 @@ class MAuthDirectivesSpec extends WordSpec with Matchers with ScalatestRouteTest
+        responseAs[String] shouldBe AuthHeaderDetail(MAuthVersion.MWS, appUuid, signature).toString
+      }
+    }
+
+    "reject with a MalformedHeaderRejection if Authentication is missing the Prefix MWS" in {
+      val wrongHeader = s" $appUuid:$signature"
+      Get().withHeaders(RawHeader(MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME, wrongHeader)) ~> route ~> check {
+        inside(rejection) {
+          case MalformedHeaderRejection(actualHeader, actualMsg, _) ⇒
+            actualHeader shouldBe "x-mws-authentication"
+            actualMsg shouldBe s"x-mws-authentication header supplied with bad format: [$wrongHeader]"
+        }
+      }
+    }
+
+    "reject with a MalformedHeaderRejection if Authentication is missing the App UUID" in {
+      val wrongHeader = s"$authPrefix :$signature"
+      Get().withHeaders(RawHeader(MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME, wrongHeader)) ~> route ~> check {
+        inside(rejection) {
+          case MalformedHeaderRejection(actualHeader, actualMsg, _) ⇒
+            actualHeader shouldBe "x-mws-authentication"
+            actualMsg shouldBe s"x-mws-authentication header supplied with bad format: [$wrongHeader]"
+        }
+      }
+    }
+
+    "reject with a MalformedHeaderRejection if Authentication is missing the signature" in {
+      val wrongHeader = s"$authPrefix $appUuid:"
+      Get().withHeaders(RawHeader(MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME, wrongHeader)) ~> route ~> check {
+        inside(rejection) {
+          case MalformedHeaderRejection(actualHeader, actualMsg, _) ⇒
+            actualHeader shouldBe "x-mws-authentication"
+            actualMsg shouldBe s"x-mws-authentication header supplied with bad format: [$wrongHeader]"
+        }
+      }
+    }
+
+    "reject with a MissingHeaderRejection if header is missing" in {
+      Get() ~> route ~> check {
+        inside(rejection) {
+          case MissingHeaderRejection(headerName) ⇒ headerName.replaceAll("_", "-").toLowerCase shouldEqual `X-MWS-Authentication`.name
+        }
+      }
+    }
+  }
+
+  "authenticate for V2" should {
+    lazy val route: Route = authenticate(executor, authenticatorV2, timeout, requestValidationTimeout).apply(complete(HttpResponse()))
+    val publicKey = MAuthKeysHelper.getPublicKeyFromString(FixturesLoader.getPublicKey)
+
+    "pass successfully authenticated request with parameters" in {
+      (client.getPublicKey _).expects(appUuid).returns(Future(Some(publicKey)))
+      //noinspection ConvertibleToMethodValue
+      (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(timeHeader)
+
+      Get("/").withHeaders(
+        RawHeader(MAuthRequest.MCC_TIME_HEADER_NAME, timeHeader.toString),
+        RawHeader(MAuthRequest.MCC_AUTHENTICATION_HEADER_NAME, authHeaderV2),
+        RawHeader(MAuthRequest.X_MWS_TIME_HEADER_NAME, timeHeaderV2.toString),
+        RawHeader(MAuthRequest.X_MWS_AUTHENTICATION_HEADER_NAME, authHeader)
```

Comment:
Maybe there should be a separate test for  V2 headers and V1 headers

Reply:
This test is to verify the client works well with the both headers of V1 and V2 provided

Reply:
Oh, I see, it is because the signer returns the two versions of headers, do you think it worth to test all the possible valid cases,   both V1 and V2 passed, only V1 and only V2?

Reply:
added more  valid cases as suggested